lib/ansible/modules/known_hosts.py

```diff
@@ -310,6 +310,8 @@ def normalize_known_hosts_key(key):
     absent in known_hosts files)
     """
     key = key.strip()  # trim trailing newline
+    if "\n" in key or "\r" in key:
+        module.fail_json(msg="Argument 'key' contains newlines. This module only accepts a single key.")
```

Comment:
just rstrip `\r` from the key after split, there is no case in which `\n` should make it to this point as we split on it before calling the function. No need for errors, `\r` is invalid in the middle of the key but it is probably part of the line break if it was saved from a Windows machine, that case we can optimize for, any other case should fail on key/file validation

Reply:
> this is not something we should validate this way, we either do full 'correct key format' validation or leave as is

Sure, I'm up for implementing full validation. But out of curiosity, why? What's wrong with a simple check for a footgun *(specifically passing multiple keys)* if you know it's never valid?

Reply:
it  is piecemeal, we would keep adding each case as it appears and make a tangle of the code, while a full format validation should cover all cases off the bat in a much cleaner way

Reply:
Fair!
I'll mark this as draft and start working on full validation. I will also take a look at other issues to see if they could be solved at once.